### PR TITLE
Auth0 deprecation

### PR DESCRIPTION
**Pull request by TitusEntertainment**
With this PR I introduce:

Login with Slack
Login with Google
Login with basic auth

+ Various bug fixes and extra stuff

### CODE REVIEW

File: src/lib/strategies/local.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { getDB } from '@database/db';
+import bcrypt from 'bcrypt';
+import { Strategy } from 'passport-local';
+
+export const setupLocalStrategy = (): Strategy => {
+	const localStrategy: Strategy = new Strategy(async (username: string, password: string, done: any) => {
+		const user = await getDB().users.findOne({ email: username.trim().toLowerCase() });
+		if (!user) return done(null, false, { msg: 'Invalid password or email' });
+		const validPassword = await bcrypt.compare(password, user.password);
```
Comment (binaryoverload) [inline src/lib/strategies/local.ts:9]: Trim password before checking,

File: src/routes/user/helpers/getSafeUser.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Request } from 'express';
+
+export const getSafeUser = (req: Request) => {
```
Comment (binaryoverload) [inline src/routes/user/helpers/getSafeUser.ts:3]: Document what this function is for! :)

**Review by binaryoverload — CHANGES_REQUESTED** [review]
Run a tab to spaces conversion on everything! :)

File: src/routes/user/controllers/createLocalUser.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+	const userExist = await getDB().users.findOne({ email: data.email });
+	if (userExist) {
+		return res
+			.status(400)
+			.send({ ...responseGenerator(400, 'There is already a user with that email') });
+	}
+	const salt = await bcrypt.genSalt(12);
+
+	const password = await bcrypt.hash(data.password, salt);
```
[the remark this answers is not in the document]
Comment (TitusEntertainment) [reply src/routes/user/controllers/createLocalUser.ts:24]: The middleware prior does this for me

File: src/lib/strategies/local.ts (hunk 1 above)
Comment (TitusEntertainment) [reply src/lib/strategies/local.ts:9]: Fixed.

**Pull request merged**